src/ansible_compat/runtime.py

```diff
@@ -112,16 +116,29 @@ def clean(self) -> None:
         if self.cache_dir:
             shutil.rmtree(self.cache_dir, ignore_errors=True)
 
-    # pylint: disable=no-self-use
-    def exec(self, args: Union[str, List[str]]) -> CompletedProcess:
-        """Execute a command inside an Ansible environment."""
-        return subprocess.run(
-            args,
-            universal_newlines=True,
-            check=False,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-        )
+    def exec(
+        self, args: Union[str, List[str]], retry: bool = False
+    ) -> CompletedProcess:
+        """Execute a command inside an Ansible environment.
+
+        :param retry: Retry network operations on failures.
```

Comment:
nit: why isn't `args` documented?

Reply:
I was not able to find a nice way to document the args/kwargs on run, as we mainly pass them to `subprocess.run` but I agree that we should expose those that matter.